File: Learning/Solns/HealthFood.java
import java.util.Arrays;

public class HealthFood {
    public static int[] selectMeals(int[] protein, int[] carbs, int[] fat, String[] dietPlans) {
        int result[] = new int[dietPlans.length];
        for (int i = 0; i < result.length; i++) {
            for (int j = 1; j < protein.length; j++) {
                if (bestMeal(protein, carbs, fat, dietPlans[i], j, result[i])) {
                    result[i] = j;
                }
            }
        }
        return result;
    }

    static boolean bestMeal(int protein[], int carbs[], int fat[], String dietPlan, int a, int b) {
        for (int i = 0; i < dietPlan.length(); i++) {
            char character = dietPlan.charAt(i);
            if (character == 'C' || character == 'c') {
                if (carbs[a] == carbs[b]) {
                    continue;
                }
                if (character == 'C') {
                    return carbs[a] > carbs[b];
                } else {
                    return carbs[a] < carbs[b];
                }
            } else if (character == 'P' || character == 'p') {
                if (protein[a] == protein[b]) {
                    continue;
                }
                if (character == 'P') {
                    return protein[a] > protein[b];
                } else {
                    return protein[a] < protein[b];
                }
            } else if (character == 'F' || character == 'f') {
                if (fat[a] == fat[b]) {
                    continue;
                }
                if (character == 'F') {
                    return fat[a] > fat[b];
                } else {
                    return fat[a] < fat[b];
                }
            } else {
                int calorie1 = protein[a] * 5 + carbs[a] * 5 + fat[a] * 9;
                int calorie2 = protein[b] * 5 + carbs[b] * 5 + fat[b] * 9;
                if (calorie1 == calorie2) {
                    continue;
                }
                if (character == 'T') {
                    return calorie1 > calorie2;
                } else {
                    return calorie1 < calorie2;
                }
            }
        }
        return false;
    }

    public static void main(String[] args) {
        int[] p = { 3, 4, 1, 5 };
        int[] c = { 2, 8, 5, 1 };
        int[] f = { 5, 2, 4, 4 };
        String[] d = { "tFc", "tF", "Ftc" };
        System.out.println(Arrays.toString(selectMeals(p, c, f, d)));
    }
}

File: CodeDump/HealthFood.js
var d = ["tFc", "tF", "Ftc"];
var p = [3, 4, 1, 5];
var c = [2, 8, 5, 1];
var f = [5, 2, 4, 4];
var result = [];
for (var i = 0; i < d.length; i++) {
  for (var j = 0; j < p.length; j++) {
    if (selectMeals(d[i], p, c, f, i, j)) {
      result[i] = j;
    }
  }
}
console.log(result);

function selectMeals(d, p, c, f, i, j) {
  for (var a = 0; a < d.length; a++) {
    if (d[i] == "C" || d[i] == "c") {
      if (c[i] == c[j]) {
        continue;
      }
      if (d[i] == "C") {
        return c[i] > c[j];
      } else {
        return c[i] < c[j];
      }
    } else if (d[i] == "P" || d[i] == "p") {
      if (d[i] == d[j]) {
        continue;
      }
      if (d[i] == "P") {
        return p[i] > p[j];
      } else {
        return p[i] < p[j];
      }
    } else if (d[i] == "F" || d[i] == "f") {
      if (f[i] == f[j]) {
        continue;
      }
      if (d[i] == "F") {
        return f[i] > f[j];
      } else {
        return f[i] < f[j];
      }
    } else {
      var cal1 = p[i] * 5 + c[i] * 5 + f[i] * 9;
      var cal2 = p[j] * 5 + c[j] * 5 + f[j] * 9;
      if (cal1 == cal2) {
        continue;
      }
      if (d[i] == "T") {
        return cal1 > cal2;
      } else {
        return cal2 < cal1;
      }
    }
  }
  return false;
}
